float64 mx
float64 my
float64 fz
